server/src/main/scala/org/broadinstitute/dsp/workbench/welder/server/ObjectService.scala

```diff
@@ -99,6 +99,7 @@ class ObjectService(
   //
   def acquireLock(req: AcquireLockRequest, traceId: TraceId): IO[Unit] =
     for {
+      _ <- permits.acquire
```

Comment:
It feels too conservative to effectively set a global parallelism of 1 across the 3 primary methods in Welder; having multiple tabs is definitely a legitimate use case. I would recommend only acquiring locks per-notebook-path - via a map of semaphores or some other alternate library.